### PR TITLE
Feature display informative fields

### LINKED ISSUE #1277: Show teacher information field when user has teacher permissions
Currently, the teacher information field is never shown, but it is intended to display information for `teacher` and `headmaster` roles

### PR DESCRIPTION
**Pull request by flbulgarelli**
Fixes #1277 
Related to https://github.com/mumuki/mumuki-domain/issues/14 
Fixes https://github.com/mumuki/mumuki-domain/issues/13

### CODE REVIEW

**Comment by flbulgarelli** [comment]
Guide with sources: 

image

**Comment by flbulgarelli** [comment]
For the teacher_info I am not completely sure about how to show it. 

We could display it this way: 

image

But we could also use student/teacher tabs on the description. 

Take into account that teacher_info is also present on exercises, where there is currently a los of information.

**Comment by flbulgarelli** [comment]
:up:

With tabs it may look the following way - ignore padding issues:

# Anonymous user
 
image

# Teacher user 

image

image

**Comment by flbulgarelli** [comment]
New screenshots for teacher_info

image


image

**Comment by JessSaavedra** [comment]
It could work, but i'd try:

- add more padding to the title
- be able to close by clicking outside the popover
- improve style by filling the tip 
- use mumuki colors in the title bar

**Comment by JessSaavedra** [comment]
Also, please try the following icons

- chevron-down and friends
- info-circle
- plus-square

**Comment by flbulgarelli** [comment]
> add more padding to the title

 Done

> improve style by filling the tip
> use mumuki colors in the title bar

Instead of that, I removed the grey color and increment thickness of font title, to resemble the corollary box. 

image
 
image

**Comment by flbulgarelli** [comment]
Finally, I am not really convinced how the learn_more field should look like. There is consensus that it should be next to the corollary, and it has no sense to have a lean-more when a corollary does not extist, but the problem is that corollaries actually have no title, they are just a final text that appear at the end of the guide, so adding a different title would not look well. 

I think that learn-more texts should be inline in the corollary rather than after it, and I am starting to think they should be a content-convention instead of a field on its own. Perhaps they could be like the description's task - the part that is quoted with `>` - simple convention for corollaries. 

It could look like the following: 

image

I think this requires further discussion, so I will remove this issue from this PR.

**Comment by flbulgarelli** [comment]
Final look: 

image


image

File: app/assets/javascripts/application/popover.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+mumuki.load(() => {
+  let $popover = $('.mu-popover');
+
+  $('html').click(function(e) {
```
Comment (julian-berbel) [inline app/assets/javascripts/application/popover.js]: This causes that even if you click _within_ the popover, it still closes, which seems odd.

Comment (flbulgarelli) [reply app/assets/javascripts/application/popover.js]: Done

**Comment by flbulgarelli** [comment]
> Trying to actually add a link somewhere seems to break the display of the popover:
> image
>

I have just fixed the escapes issue. 
 
> Also, I'm just nitpicking here, but it bothers me a little that the tooltip overlaps with the popover's title when you click it:
> image
> 
> Feel free to ignore these if you think they're not important.

I agree, but I don't find an easy way of solving that

**Pull request merged**